update-repo.py

```diff
@@ -0,0 +1,140 @@
+import subprocess
+import sys
+import os
+
+import shutil
+
+
+"""
+This script streamlines the process of copying a .yml file into each workflows directory, 
+followed by creating a PR into the GitHub feedstock repository.
+
+How to use:
+
+python /path/.../update-repo.py <filepath-you-want-to-copy-in-your-local-machine> <filepath-in-the-github-repo>
+
+Workflow:
+
+- Walk through all directories to find corresponding filepath
+- Copy file to desired directories
+- Fetch the user's GitHub username using the GitHub CLI for authentication
+- Commit these changes, pushes them to GitHub, and creates a PR
+"""
+
+"""
+Utility Functions
+"""
+
+
+def run_command(command, cwd=None):
+    """Run a shell command in the specified directory."""
+    subprocess.run(
+        command,
+        check=True,
+        shell=True,
+        text=True,
+        cwd=cwd,
+    )
+
+
+"""
+Core Functionalities - find all desired directories, copy desired file, and create PR
+"""
+
+
+def copy_file(source_file, root_dir, dst_dir, username):
+    for dirpath, dirnames, filenames in os.walk(root_dir):
+        if dst_dir in dirnames:
+            workflow_dir = os.path.join(dirpath, str(dst_dir))  # Full path to the workflow directory
+            dest_file = os.path.join(workflow_dir, source_file)  # Destination file path
+
+            # Check if the file already exists
+            if not os.path.exists(dest_file):
+                # Check out main and pull the latest changes
+                run_command("git checkout main", cwd=dirpath)
+                run_command("git pull upstream main", cwd=dirpath)
+
+                # Create and switch to a new branch named after the new version
+                run_command(f"git checkout -b {source_file}", cwd=dirpath)
+
+                # Copy the file
+                shutil.copy2(source_file, dest_file)
+                print(f'File copied to {workflow_dir}')
+
+                # Create PR
+                package_name = dirpath.split('/')[1]
+                if package_name == "bg-mpl-stylesheets":
```

Comment:
packages under diffpy.py all start with diffpy (for the ones that we maintain), 

<img width="256" alt="Screenshot 2024-10-05 at 10 49 33 PM" src="https://github.com/user-attachments/assets/ea19e60d-33f9-47d3-bf82-0758c472089e">

so, let's improve the logic with that if the string starts with diffpy, then org becomes `diffpy`, otherwise, `Billinge group`